File: Python Codes/USPMO.py
def digitSum(num):
	digit=0
	sum=0
	num=num//10
	for i in range(10):
		digit=num%10
		num=num//10
		sum=sum+digit
	return sum;

def isValidUSPMO(uspmo):
	if(len(uspmo)!=11):
		return False
	uspmo=int(uspmo)
	if(digitSum(uspmo)%9==uspmo%10):
		return True
	else:	
		return False

def errorRectifier(num):
	num=int(num)
	print("Possible Valid Code can be (Only if the Last Digit is entered incorrectly):",(num//10)*10 +(digitSum(num)%9))

code=input("Enter a US Postal Money Order Code: ")
if isValidUSPMO(code):
	print("Valid")
else:
	print("Invalid ")
	errorRectifier(code);
	

File: Cpp Codes/USPMO.cpp
#include <iostream>
using namespace std;

int digitSum(long num)
{
    int digit = 0, sum = 0;
    num = num / 10;
    for (int i = 0; i < 10; i++)
    {
        digit = num % 10;
        num = num / 10;
        sum += digit;
    }
    return sum;
}

int isValidUSPMO(long uspmo)
{
    string uspmo_str = to_string(uspmo);
    if (uspmo_str.length() != 11)
        return false;
    if (digitSum(uspmo) % 9 == uspmo % 10)
        return true;
    else
        return false;
}

int errorRectifier(long num)
{
    cout << "Possible Valid Code can be (Only if the Last Digit is entered incorrectly): "
         << (num / 10) * 10 + (digitSum(num) % 9) << endl;
    return 0;
}
int main()
{
    long code;
    cout << "Enter a US Postal Money Order Code: ";
    cin >> code;
    if (isValidUSPMO(code))
        cout << "Valid" << endl;
    else
    {
        cout << "Invalid" << endl;
        errorRectifier(code);
    }
    return 0;
}